optional/src/main/java/org/dmfs/optional/single/Optimistic.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.optional.single;
+
+import org.dmfs.jems.single.Single;
+import org.dmfs.optional.Optional;
+
+import java.util.NoSuchElementException;
+
+
+/**
+ * An optimistic {@link Optional} to {@link Single} adapter which assumes the value is present.
+ * <p>
+ * Note: if the {@link Optional} value is not present {@link #value()} will fail with a {@link NoSuchElementException}.
+ *
+ * @author Marten Gajda
+ */
+public final class Optimistic<T> implements Single<T>
```

Comment:
So what I mean is the thing that I mentioned many times now, that we should never call `optional.value()` without calling `isPresent()` first, that's clearly wrong, it's not an `Optional` at that point. And using this class with this name sort of encourages that for me.
I see it's used inside another class, maybe it could be an inner one there.

Or maybe it can work as public with a different name.

Reply:
I'll remove it for now. I've added it so support `Lazy` but I'm going to remove that too.